libvmaf/test/test_predict.c

```diff
@@ -49,14 +57,120 @@ static char *test_predict_score_at_index()
 
     double vmaf_score = 0.;
     err = vmaf_predict_score_at_index(model, feature_collector, 0, &vmaf_score,
-                                      true, 0);
+                                      true, false, 0);
     mu_assert("problem during vmaf_predict_score_at_index", !err);
 
     vmaf_model_destroy(model);
     vmaf_feature_collector_destroy(feature_collector);
     return NULL;
 }
 
+
+void set_meta(void *data, const char *key, double d)
+{
+    if (!data) return;
+    MetaStruct *meta = data;
+    char value[128];
+    snprintf(value, sizeof(value), "%f", d);
+    vmaf_dictionary_set(meta->metadata, key, value, meta->flags);
+}
+
+static char* test_propagate_metadata()
+{
+    int err;
+
+    VmafDictionary *dict = NULL;
+    MetaStruct meta_data = {
+        .metadata = &dict,
+        .flags    = 0,
+    };
+
+    VmafMetadataConfig m = {
+        .callback = set_meta,
+        .data     = &meta_data,
+        .data_sz  = sizeof(meta_data),
```

Comment:
Doesn't seem like `data_sz` is used anywhere. Since this API should not be in charge of allocating `data`, I don't think we need this.